notification-manager/build.gradle.kts

```diff
@@ -4,20 +4,17 @@ plugins {
 	id("com.qpay.java.base")
 }
 
-java {
-	sourceCompatibility = JavaVersion.VERSION_21
-}
-
-configurations {
-	compileOnly {
-		extendsFrom(configurations.annotationProcessor.get())
-	}
-}
-
 dependencies {
 	implementation(project(":libs"))
 	implementation("org.springframework.boot:spring-boot-starter-web")
 	implementation("org.springframework.boot:spring-boot-starter-validation")
 	implementation("org.springframework.boot:spring-boot-starter-mail")
 	implementation("org.springframework.kafka:spring-kafka:${Versions.KAFKA}")
+	implementation("org.springframework.cloud:spring-cloud-starter-netflix-eureka-client")
+}
+
+dependencyManagement {
+	imports {
+		mavenBom("org.springframework.cloud:spring-cloud-dependencies:${Versions.SPRING_CLOUD}")
+	}
```

Comment:
since you use everywhere spring boot web and spring cloud (or maybe some other spring libs), it makes sense to create a new plugin `spring.base` which includes `java.base` and these common dependencies.
For the libs module you don't need to use a new plugin, but others do